vyper/ast/parse.py

```diff
@@ -1,14 +1,112 @@
 import ast as python_ast
 import tokenize
 from decimal import Decimal
-from typing import Optional, cast
+from typing import Any, Dict, List, Optional, Union, cast
 
 import asttokens
 
-from vyper.exceptions import CompilerPanic, SyntaxException
+from vyper.ast import nodes as vy_ast
+from vyper.ast.pre_parser import pre_parse
+from vyper.compiler.settings import Settings
+from vyper.exceptions import CompilerPanic, ParserException, SyntaxException
 from vyper.typing import ModificationOffsets
 
 
+def parse_to_ast(*args: Any, **kwargs: Any) -> vy_ast.Module:
+    return parse_to_ast_with_settings(*args, **kwargs)[1]
+
+
+def parse_to_ast_with_settings(
+    source_code: str,
+    source_id: int = 0,
+    module_path: Optional[str] = None,
+    resolved_path: Optional[str] = None,
+    add_fn_node: Optional[str] = None,
+) -> tuple[Settings, vy_ast.Module]:
+    """
+    Parses a Vyper source string and generates basic Vyper AST nodes.
+
+    Parameters
+    ----------
+    source_code : str
+        The Vyper source code to parse.
+    source_id : int, optional
+        Source id to use in the `src` member of each node.
+    contract_name: str, optional
+        Name of contract.
+    add_fn_node: str, optional
+        If not None, adds a dummy Python AST FunctionDef wrapper node.
```

Comment:
it's used for constructing special builtins - https://github.com/vyperlang/vyper/blob/10564dcc37756f3d3684b7a91fd8f4325a38c4d8/vyper/builtins/_utils.py#L17

(this machinery should go away soon now btw that we can implement a stdlib using native vyper instead of being compiler intrinsics)